src/styles/components/main.scss

```diff
@@ -0,0 +1,26 @@
+.main {
+  display: grid;
+  padding: 50px 40px;
+  grid-template-columns: repeat(4, 200px);
+  gap: 46px 48px;
+  justify-content: center;
+  box-sizing: border-box;
+}
+
+@media (max-width: 1023px) {
+  .main {
+    grid-template-columns: repeat(3, 200px);
+  }
+}
+
+@media (max-width: 677px) {
+  .main {
+    grid-template-columns: repeat(2, 200px);
+  }
+}
+
+@media (max-width: 487px) {
+  .main {
+    grid-template-columns: repeat(1, 200px);
```

Comment:
Replace the fixed `200px` with `1fr` to ensure that the single column takes up the full width of the container on the smallest screens. For instance, `grid-template-columns: 1fr;`.